### PR TITLE
docs: nicer documentation setup

### PR DESCRIPTION
**Pull request by avik-pal**
Preview: https://enzymead.github.io/Reactant.jl/previews/PR284/?pr=284

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -1,13 +1,84 @@
...
+
+  - icon: 🧩
+    title: TODO
+    details: TODO
+    link: TODO
+
+  - icon: 🧪
+    title: TODO
+    details: TODO
+    link: TODO
```
Comment (avik-pal) [inline docs/src/index.md:43]: This is where we can add attributes like https://lux.csail.mit.edu/stable/. Let me know what you want me to add here

**Comment by wsmoses** [comment]
this is 12/10

**Review by wsmoses — APPROVED** [review]

File: docs/src/index.md (hunk 1 above)
Comment (wsmoses) [reply docs/src/index.md:43]: Maybe Fast, Easy to Use, Interoperable/Composable?

I'm not particularly creative though if @mofeing @Pangoraw @jumerckx or others have thoughts on

Comment (jumerckx) [reply docs/src/index.md:43]: Maybe something along the lines of:
- Best of Both Worlds
	Reactant combines Julia's high-level syntax/language features with MLIR/XLA's state-of-the-art compiler
- GPU-friendly
	Reactant can run your code on CPU, GPU, and more.
Also just riffing, there's surely better attributes as well.

Comment (wsmoses) [reply docs/src/index.md:43]: Yeah having something that alludes to the fact that you can write regular julia code and automatically and efficiently run it on any (eventually combination of) backends would be great

Comment (mofeing) [reply docs/src/index.md:43]: mmm maybe we can add some interesting points like...
- "No need for `vmap`: Broadcasting is batched!"
- "Directly differentiate your MLIR code thanks to Enzyme"

Comment (wsmoses) [reply docs/src/index.md:43]: Re the latter bullet maybe the keyword is differetentiable and that’s the description

Comment (avik-pal) [reply docs/src/index.md:43]: We should mention something along "Batteries included with MLIR AD"

Comment (wsmoses) [reply docs/src/index.md:43]: The fancy optimizations in Enzyme-JaX don't yet have their own name, but something referring to said optimizations would be good

Comment (mofeing) [reply docs/src/index.md:43]: yeah, my point is that it's cool to say that we're differentiating the MLIR directly, unlike JAX, torch, ... (although yeah, JAX can use Enzyme-JAX for that) which differentiate the DAG

File: docs/src/components/AsideTrustees.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <h2> Trusted by </h2>
+    <a class="enjoyer" href="https://lux.csail.mit.edu/" target="_blank">
+        <img width="32" height="32" src="https://raw.githubusercontent.com/LuxDL/Lux.jl/refs/heads/main/assets/lux-logo.svg" />
+        <span>
+            <p class="extra-info">Scientific Computing</p>
+            <p class="heading">Lux.jl</p>
+            <p class="extra-info">Machine Learning</p>
+        </span>
+    </a>
+    <!-- XXX: Add more projects using Reactant.jl here -->
```
Comment (avik-pal) [inline docs/src/components/AsideTrustees.vue:13]: Add projects here makes it show up on the side banner

Comment (avik-pal) [reply docs/src/components/AsideTrustees.vue:13]: can you give 2 lines for the extra-info about tenet

Comment (mofeing) [reply docs/src/components/AsideTrustees.vue:13]: do you mean like a small description of Tenet? or the description of the integration?

Comment (mofeing) [reply docs/src/components/AsideTrustees.vue:13]: I see. I would say:

1. "Quantum Simulation"
2. "Tensor Networks"

Comment (avik-pal) [inline docs/src/components/AsideTrustees.vue:13]:
```suggestion
    <a class="enjoyer" href="https://bsc-quantic.github.io/Tenet.jl/stable/" target="_blank">
        <img width="32" height="32" src="https://raw.githubusercontent.com/bsc-quantic/Tenet.jl/refs/heads/master/docs/src/assets/logo.svg" />
        <span>
            <p class="extra-info">Quantum Simulation</p>
            <p class="heading">Tenet.jl</p>
            <p class="extra-info">Tensor Networks</p>
        </span>
    </a>
```

File: docs/src/index.md (hunk 1 above)
Comment (avik-pal) [reply docs/src/index.md]: I have added an initial iteration of the captions

**Comment by wsmoses** [comment]
Looks great to me! Feel free to merge whenever you like (and we will continue to add to in subsequent PRs of course)

**Comment by wsmoses** [comment]
Minor comment,  but could we put the arrays in the code reactant API section

**Comment by avik-pal** [comment]
> Minor comment, but could we put the arrays in the code reactant API section

Those are currently undocumented, so CI will fail. To keep this PR manageable I avoided adding any additional docstrings (even `@compile` is mostly empty). I will follow this up with docstrings

**Pull request merged**